gui/confirm.lua

```diff
@@ -1,8 +1,36 @@
 -- config ui for confirm
+--@ module = true
+local GLOBAL_KEY = 'confirm'
+local CONFIRM_CONFIG_FILE = 'dfhack-config/confirm.json'
+local json = require('json')
+confirm = require('plugins.confirm')
+gui = require('gui')
+widgets = require('gui.widgets')
 
-confirm = require 'plugins.confirm'
-gui = require 'gui'
-widgets = require 'gui.widgets'
+loaded = loaded or false
+
+if dfhack.onStateChange then
+    dfhack.onStateChange[GLOBAL_KEY] = function(_)
+        if loaded then
+            return
+        end
+
+        local ok, config = pcall(json.decode_file, CONFIRM_CONFIG_FILE)
+        local conf_data = ok and config or confirm.get_conf_data()
+        if ok then
+            for _, c in ipairs(conf_data) do
+                -- for some reason changing the conf state here doesn't seem to change the result
+                -- when the dialog is opened like it did when it was in the other script?
+                confirm.set_conf_state(c.id, c.enabled)
+            end
+        end
+        loaded = true
+    end
+end
+
+if dfhack_flags.module then
+    return
+end
```

Comment:
this block should go at the end of the file, just before the last line that instantiates the UI